### PR TITLE
Docs: Initial overhaul for stable in 2024 (GameIndex)

### PR DESCRIPTION
**Pull request by RedDevilus**
### Description of Changes

Adds some new values and more contextual extra info. Needed for having up-to date information. Still need to update other docs like FAQ, Readme and Debugger but that's future work.
### Rationale behind Changes

They were due 2 years.
### Suggested Testing Steps

Check your docs folder on respective OS.

### CODE REVIEW

**Review by Mrlinkwii — APPROVED** [review]

**Review by kamfretoz — APPROVED** [review]

File: pcsx2/Docs/GameIndex.md [hunk 1]
```diff
@@ -136,124 +138,131 @@ The clamp modes are also numerically based.
...
+  * Checks for possible float overflows (Superman Returns).
 
-*   `VUOverflowHack`
-    *   VU Overflow hack to check for possible float overflows (Superman Returns).
+* `VUSyncHack`
+  * Ensures synchronization between VUs and EE to fix timing issues.
 
-*   `SoftwareRendererFMVHack`
-    *   Forces rendering into software mode during FMVs.
+* `XgKickHack`
```
Comment (refractionpcsx2) [inline pcsx2/Docs/GameIndex.md:239]: G should be capital, like it was before

File: pcsx2/Docs/GameIndex.md [hunk 2]
```diff
@@ -136,124 +138,131 @@ The clamp modes are also numerically based.
...
+* bilinearUpscale            [`0` or `1` or `2`]                        {Automatic, Force Bilinear, Force Nearest}     Default: Automatic // Tab 4 Upscaling Fixes (3rd dropdown)
+* skipDrawStart              [Value between `0` to `10000`]             {0-10000}                          Default: Off (`0`) // Tab 4 Upscaling Fixes (4th position spinbox left)
+* skipDrawEnd                [Value between `0` to `10000`]             {0-10000}                          Default: Off (`0`) // Tab 4 Upscaling Fixes (4th position spinbox right)
+* halfPixelOffset            [`0` or `1` or `2` or `3` or `4`] {Off, Normal Vertex, Special (Texture), Special (Texture Aggressive), Align to Native} Default: Off (`0`) // Tab 4 Upscaling Fixes (1st dropdown)
+* nativePaletteDraw          [`0` or `1`]           {Off, On}                               Default: Off (`0`) // Tab 4 Upscaling Fixes (1st checkbox on right row 2)
+* roundSprite                [`0` or `1` or `2`]    {Off, Half or Full}                     Default: Off (`0`) // Tab 4 Upscaling Fixes (2nd dropdown)
```
Comment (refractionpcsx2) [inline pcsx2/Docs/GameIndex.md:183]: None of these comments are required and make absolutely no sense in this file.

File: pcsx2/Docs/GameIndex.md [hunk 3]
```diff
@@ -136,124 +138,131 @@ The clamp modes are also numerically based.
...
+* gpuPaletteConversion        [`0` or `1`]          {Off, On}                               Default: Off (`0`) // Tab 3 Hardware Fixes (2nd checkbox on right row 2)
+* gpuTargetCLUT               [`0` or `1` or `2`]   {Disabled, Enabled (Exact Match), Enabled (Check Inside Target)}                     Default: Disabled (`0`) // Tab 3 Hardware Fixes (3rd dropdown)
+* minimumBlendingLevel        [`0` or `1` or `2` or `3` or `4` or `5`]      {Minimum, Basic, Medium, High, Full(Slow), Maximum (Very Slow)}    Default: Automatic (No value, looks up GameDB)
+* maximumBlendingLevel        [`0` or `1` or `2` or `3` or `4` or `5`]      {Minimum, Basic, Medium, High, Full(Slow), Maximum (Very Slow)}    Default: Automatic (No value, looks up GameDB)
+* recommendedBlendingLevel    [`0` or `1` or `2` or `3` or `4` or `5`]      {Minimum, Basic, Medium, High, Full(Slow), Maximum (Very Slow)}    Default: Automatic (No value, looks up GameDB)
+* readTCOnClose               [`0` or `1`]          {Off, On}                               Default: Off (`0`) // Tab 3 Hardware Fixes (4th checkbox on right row 2)
```
Comment (refractionpcsx2) [inline pcsx2/Docs/GameIndex.md:171]: As below, comments are not required and shouldn't be in this file.

File: pcsx2/Docs/GameIndex.md [hunk 4]
```diff
@@ -136,124 +138,131 @@ The clamp modes are also numerically based.
...
-        {None, Partial or Full Hash Cache}
-        Default: None (`0`)
-*   deinterlace
-        [Value between `0` to `9`]
-        {Automatic, Off, WeaveTFF, WeaveBFF, BobTFF, BobBFF, BlendTFF, BlendBFF, AdaptiveTFF, AdaptiveBFF}
-        Default: Automatic (No value, looks up GameDB)
+* beforeDraw                {`OI` with suffix }  {None unless specific game GSC}         Default: Automatic (No value, looks up GameDB) with valid variable name (ex. OI_BurnoutGames)
+* afterDraw                 {`OO` with suffix }  {None unless specific game GSC}         Default: Automatic (No value, looks up GameDB) with valid variable name
+* conservativeFramebuffer   [`0` or `1`]               {Off or On}                             Default: On (`1`)
+* texturePreloading         [`0` or `1` or `2`]        {None, Partial or Full Hash Cache}     Default: None (`0`) // Tab 2 Rendering (8th dropdown)
```
Comment (refractionpcsx2) [inline pcsx2/Docs/GameIndex.md:149]: Remove location comments.

**Comment by RedDevilus** [comment]
image

Non-relevant red X, because it looks at the patches.

**Review by Mrlinkwii — APPROVED** [review]

**Review by kamfretoz — APPROVED** [review]

**Comment by TheTechnician27** [comment]
This was just a quick once-over for grammatical mistakes, and I may have missed things:

* 'Scarface The World is Yours' should have a colon indicating the subtitle, namely 'Scarface: The World is Yours'.
* 'Katamari Damacy have weird speed bug' should be 'has a weird speed bug'.
* 'T-bit games dislike MTVU and some games' is using 'and' as a conjunction and thus needs a comma.
* 'Parappa' is actually spelled with camel-case, namely 'PaRappa'.
* 'for which this override is for' has a redundant 'for'. This could be changed to 'for which this override is used'.
* 'nor it's save memcard' uses the wrong 'its' (contraction rather than possessive).
* 'refers to the VU's in micro mode' uses the possessive of 'VU', whereas it seems like it should be the plural 'VUs'.

I didn't include missing periods because their usage doesn't seem standardized within the document anyway. I also noticed in the PDF that some comments seem like they're supposed to be on two lines but end up in one instead. For example:

image

**Comment by RedDevilus** [comment]
> This was just a quick once-over for grammatical mistakes, and I may have missed things:
> 
>     * 'Scarface The World is Yours' should have a colon indicating the subtitle, namely 'Scarface: The World is Yours'.
> 
>     * 'Katamari Damacy have weird speed bug' should be 'has a weird speed bug'.
> 
>     * 'T-bit games dislike MTVU and some games' is using 'and' as a conjunction and thus needs a comma.
> 
>     * 'Parappa' is actually spelled with camel-case, namely 'PaRappa'.
> 
>     * 'for which this override is for' has a redundant 'for'. This could be changed to 'for which this override is used'.
> 
>     * 'nor it's save memcard' uses the wrong 'its' (contraction rather than possessive).
> 
>     * 'refers to the VU's in micro mode' uses the possessive of 'VU', whereas it seems like it should be the plural 'VUs'.
> 
> 
> I didn't include missing periods because their usage doesn't seem standardized within the document anyway. I also noticed in the PDF that some comments seem like they're supposed to be on two lines but end up in one instead. For example:
> 
> image

Probably should've checked what was already in PCSX2 itself and not just the additions, but fixed it.

**Comment by TheTechnician27** [comment]
> Probably should've checked what was already in PCSX2 itself and not just the additions, but fixed it.

I'll go ahead and take a look. At that moment, I just wanted to take a glance at the changes since link asked me, but a proper overview is a good idea.

**Comment by TheTechnician27** [comment]
* 'These values are in a key-value format' should technically use what's called an en-dash '–' instead of a hyphen '-'.
* 'You can use this to validate the file, and assist in writing it properly.' The second half isn't a complete thought and thus does not need a comma (it is incorrect with one).
* 'Both the serial numbers for the games, and the CRC patches are' This is a list of two things and thus does not need to be separated by a comma.
* 'throw a validation error and the first entry' This is using 'and' as a conjunction and thus would ideally have a comma before 'and'.
* 'and require the string to be quoted, be aware / use a YAML linter to avoid confusion.' 'Be aware' is the start of a new thought, and therefore it should be 'and require the string to be quoted. Be aware / use a YAML linter to avoid confusion.'
* 'These values are case-sensitive so take care' There should be a comma before 'so' here.
* 'Any invalid game-fixes' We use 'game fixes' throughout the documentation, except for here, where it's strangely hyphenated.
* 'impacting games like Fifa Street 2' FIFA is an acronym for a proper noun and thus is entirely uppercased.
* By fixing 'PaRappa', you inadvertently misspelled it as 'ParRappa'.
* 'multi-disc games, where later games often reuse the serial of the previous disc(s), and games' By grammatical standards I think nothing's wrong with this, but it may read better if, instead of enclosing 'where later games often reuse the serial of the previous disc(s)' in commas, you enclosed it in en-dashes. Namely:
** 'Examples include multi-disc games – where later games often reuse the serial of the previous disc(s) – and games that allow transfer'
* 'same CRC cannot be defined and this will throw a validation error.' 'and' here is being used as a conjunction and should have a comma before it.

This is everything I was able to find at the moment.

**Comment by RedDevilus** [comment]
'You can use this to validate the file, and assist in writing it properly.' from what I remember a specific style and not splitting it up in two sentences

**Comment by RedDevilus** [comment]
> * 'impacting games like Fifa Street 2' FIFA is an acronym for a proper noun and thus is entirely uppercased.
> 
> *

Problem is I have to directly modify the PDF, so it's going to mess a ton of stuff if I add the example for multi-disc in the PDF. And this was just going to be a precursor to bigger fixes. I appreciate effort but just going to change it in the md alone.

**Review by JordanTheToaster — APPROVED** [review]

**Review by Mrlinkwii — APPROVED** [review]

**Review by kamfretoz — APPROVED** [review]

**Comment by TheTechnician27** [comment]
A line was changed to: 'T-bit games dislike MTVU, some games and are incompatible with MTVU.' Should be 'T-bit games dislike MTVU, and some games are incompatible with MTVU.' After that I think it's good to go!

**Review by TheTechnician27 — APPROVED** [review]
Worth noting that the SOCOM II -> SOCOM 2 change actually is slightly incorrect as it is stylized 'SOCOM II', but that's basically whatever. All looks good to me.

**Pull request merged**